packages/expo-in-app-purchases/android/src/main/java/expo/modules/inapppurchases/BillingManager.java

```diff
@@ -143,9 +145,14 @@ public void run() {
           return;
         }
 
-        BillingFlowParams purchaseParams = BillingFlowParams.newBuilder()
-          .setSkuDetails(skuDetails).setOldSku(oldSku).build();
-        mBillingClient.launchBillingFlow(mActivity, purchaseParams);
+        BillingFlowParams.Builder purchaseParams = BillingFlowParams.newBuilder()
+          .setSkuDetails(skuDetails);
+        if (oldSku != null) {
+          purchaseParams.setSubscriptionUpdateParams(
+            BillingFlowParams.SubscriptionUpdateParams.newBuilder().setOldSkuPurchaseToken(oldSku).build()
+          );
+        }
```

Comment:
Have you tested this? Specifically, old SKU and old SKU purchase token look like two separate things.

Reply:
Fixed here- https://github.com/expo/expo/pull/13884/commits/d8c71b1206b4e68ef235f611597e55c81e1019f3 (and tested manually)

However, this now requires you to run `getPurchaseHistoryAsync` before you run `purchaseItemAsync`. Most likely people are already doing this, but it is technically a breaking change. 
If we're making a breaking change, it might be a better idea to change `purchaseItemAsync` to accept the purchaseToken as the second argument directly, instead of the old SKU. The user would get this purchase token the same way we do- by querying their purchase history.

Reply:
APIs that require you to call method A before method B while letting you write B(); A(); are usually bad, to put it simply. Lifecycles are some of the worst parts of APIs.

Your proposal to accept the purchase token instead of the SKU structurally fixes this by making B() take the result of A(), forcing the programmer to write code in that order.